plugins/Tour/Engagement/ChallengeSelectDateRange.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\Tour\Engagement;
+
+use Piwik\Piwik;
+
+class ChallengeSelectDateRange extends Challenge
```

Comment:
If I select a date range in the dashboard, the page will reload, but the task won't be finished in the widget. If I reload the widget, it will show as finished.

Reply:
Yes that's kind of expected unfortunately, hard to fix and not worth it spending too much time on it.